src/Sidekick.Business/Apis/Poe/Trade/Data/Stats/StatDataService.cs

```diff
@@ -0,0 +1,170 @@
+
+                    entry.Pattern = new Regex($"\\n+{pattern}{suffix}");
+                    patterns.Add(entry);
+                }
+            }
+        }
+
+        public Mods ParseMods(string text)
+        {
+            text = NewLinePattern.Replace(text, "\n");
+
+            var mods = new Mods();
+
+            // Make sure the text ends with an empty line for our regexes to work correctly
+            if (!text.EndsWith("\n"))
+            {
+                text += "\n";
+            }
+
+            FillMods(mods.Explicit, ExplicitPatterns, text);
+            FillMods(mods.Implicit, ImplicitPatterns, text);
+            FillMods(mods.Enchant, EnchantPatterns, text);
+            FillMods(mods.Crafted, CraftedPatterns, text);
+            // FillMods(mods.Veiled, VeiledPatterns, text);
+
+            return mods;
+        }
+
+        private void FillMods(List<Mod> mods, List<StatData> patterns, string text)
+        {
+            var results = patterns
+                .Where(x => x.Pattern.IsMatch(text))
+                .ToList();
+
+            foreach (var x in results)
+            {
+                var result = x.Pattern.Match(text);
+                var magnitudes = new List<Magnitude>();
+
+                if (result.Groups.Count > 1)
+                {
+                    for (var index = 1; index < result.Groups.Count; index++)
+                    {
+                        double? value = null;
+                        if (double.TryParse(result.Groups[index].Value, out var parsedValue))
+                        {
+                            value = parsedValue;
+                        }
+                        magnitudes.Add(new Magnitude()
+                        {
+                            Hash = x.Id,
+                            Max = value,
+                            Min = value,
+                        });
+                    }
+                }
+                else
+                {
+                    magnitudes.Add(new Magnitude()
+                    {
+                        Hash = x.Id
+                    });
+                }
+
+                mods.Add(new Mod()
+                {
+                    Magnitudes = magnitudes
+                });
+            }
+        }
+
+        public StatData GetById(string id)
+        {
+            if (ImplicitPatterns.Any(x => x.Id == id))
+            {
+                return ImplicitPatterns.First(x => x.Id == id);
+            }
+            if (ExplicitPatterns.Any(x => x.Id == id))
+            {
+                return ExplicitPatterns.First(x => x.Id == id);
+            }
+            if (CraftedPatterns.Any(x => x.Id == id))
+            {
+                return CraftedPatterns.First(x => x.Id == id);
+            }
+            if (EnchantPatterns.Any(x => x.Id == id))
+            {
+                return EnchantPatterns.First(x => x.Id == id);
+            }
+            return null;
```

Comment:
Doing `.Any()` and then `.First()` is a bit of a weird pattern and not very efficient. Would like to see a different solution.